packages/ckeditor5-list/src/liststylescommand.js

```diff
@@ -0,0 +1,206 @@
+/**
+ * @license Copyright (c) 2003-2020, CKSource - Frederico Knabben. All rights reserved.
+ * For licensing, see LICENSE.md or https://ckeditor.com/legal/ckeditor-oss-license
+ */
+
+/**
+ * @module list/liststylescommand
+ */
+
+import Command from '@ckeditor/ckeditor5-core/src/command';
+import TreeWalker from '@ckeditor/ckeditor5-engine/src/model/treewalker';
+
+/**
+ * The list style command. It is used by the {@link module:list/liststyles~ListStyles list styles feature}.
+ *
+ * @extends module:core/command~Command
+ */
+export default class ListStylesCommand extends Command {
+	/**
+	 * Creates an instance of the command.
+	 *
+	 * @param {module:core/editor/editor~Editor} editor The editor instance.
+	 * @param {String} defaultType The list type that will be used by default if the value was not specified during
+	 * the command execution.
+	 */
+	constructor( editor, defaultType ) {
+		super( editor );
+
+		/**
+		 * The default type of the list style.
+		 *
+		 * @private
+		 * @member {String}
+		 */
+		this._defaultType = defaultType;
+	}
+
+	/**
+	 * @inheritDoc
+	 */
+	refresh() {
+		this.value = this._getValue();
+		this.isEnabled = this._checkEnabled();
+	}
+
+	/**
+	 * Executes the command.
+	 *
+	 * @param {Object} options
+	 * @param {String|null} options.type The type of the list styles, e.g. 'disc' or 'square'. If `null` specified, the default
+	 * style will be applied.
+	 * @protected
+	 */
+	execute( options = {} ) {
+		const model = this.editor.model;
+		const document = model.document;
+		const position = findPositionInsideList( document.selection );
```

Comment:
This might cause "single item changed on range selection" issues.

From top of my head:

1.  Get elements/blocks from selection (to would get all selected list items).
2.  Use `selection.getFirst/LastPostion()` to get backward/forward siblings.